### PR TITLE
[BORG UPDATE] cyborg storage upgrade

### PR DESCRIPTION
**Pull request by PhantornRU**
## What Does This PR Do

Port of the refactor and new tools from "SS220 WyccStation".
We played with these changes for 2 years and don’t touch them anymore, since we fully tested them for balance, usability and user response. We hope you will like them too and improve the gameplay of robots.

- Improved ability to carry more materials via upgrade.

## Why It's Good For The Game

Late game unit development.

## Images of changes

image

## Testing

Tested for 2 years on the SS220 WyccStation on live servers.

## Changelog
:cl:
add: Added Cyborg storage upgrade
/:cl:

### CODE REVIEW

**Comment by matttheficus** [comment]
Does this simply allow them to "charge up" more metal/glass when theyre in a recharger?

If so, we're likely looking to move EngiBorgs toward requiring to resupply at an ORM, so with that in mind, not really a fan of giving them MORE metal for "free".

**Comment by matttheficus** [comment]
#23944 is the direction I'm referring to.

**Comment by S34NW** [comment]
Although this has passed the threshold for closure with 3 Balance objections and nobody in favour, objection reasons indicate that the problem areas are mitigated by changes following https://github.com/ParadiseSS13/Paradise/pull/23944.
As such this PR will stay open beyond the usual threshold to allow Balance to re-review once https://github.com/ParadiseSS13/Paradise/pull/23944 is merged.

We will reach out to you when this is the case for further discussion.

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 1]
```diff
@@ -1159,6 +1159,16 @@
 	construction_time = 12 SECONDS
 	category = list("Cyborg Upgrade Modules")
 
+/datum/design/borg_upgrade_storage
+	name = "Cyborg Upgrade (Storage upgrade)"
+	id = "borg_upgrade_storage"
+	build_type = MECHFAB
+	build_path = /obj/item/borg/upgrade/storage
+	req_tech = list("bluespace" = 5, "materials" = 7, "engineering" = 5)
+	materials = list(MAT_METAL=15000, MAT_BLUESPACE=2000, MAT_SILVER=6000)
```
Comment (Henri215) [inline code/modules/research/designs/mechfabricator_designs.dm:1168]:
```suggestion
	materials = list(MAT_METAL = 15000, MAT_BLUESPACE = 2000, MAT_SILVER = 6000)
```

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 2]
```diff
@@ -278,6 +278,39 @@
...
+/***********************/
+
+/obj/item/borg/upgrade/storage
+	name = "storage upgrade"
+	desc = "Improves cyborg storage with bluespace technology."
+	icon_state = "cyborg_upgrade2"
+	origin_tech = "bluespace=4;materials=5;engineering=3"
+	require_module = TRUE
+	var/max_energy_multiplication = 3
+	var/recharge_rate_multiplication = 2
```
Comment (Henri215) [inline code/game/objects/items/robot/robot_upgrades.dm:292]: autodoc these

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 3]
```diff
@@ -278,6 +278,39 @@
 
 	return TRUE
 
+/***********************/
+/****	FOR ALL		****/
+/***********************/
+
+/obj/item/borg/upgrade/storage
+	name = "storage upgrade"
+	desc = "Improves cyborg storage with bluespace technology."
```
Comment (Henri215) [inline code/game/objects/items/robot/robot_upgrades.dm:287]: Improves how? The description could be expande

**Review by DGamerL — CHANGES_REQUESTED** [review]
Please do a merge master, it has a stealth conflict

**Comment by SteelSlayer** [comment]
Author does not have time to work on this at the moment. Closing for now.

image

**Pull request closed without merging**